Source/MQTTnet/Server/MqttClientSessionsManager.cs

```diff
@@ -240,10 +240,10 @@ private async Task HandleClientAsync(IMqttChannelAdapter channelAdapter, Cancell
                     return;
                 }
 
-                clientId = connectPacket.ClientId;
-
                 var connectionValidatorContext = await ValidateConnectionAsync(connectPacket, channelAdapter).ConfigureAwait(false);
 
+                clientId = connectPacket.ClientId;
```

Comment:
I am not sure if this is correct. The Client ID from a CONNECT packet might be null. But the assigned client ID is set in another property. Isn't it required to use either ClientId OR AssignedClientId (in can the first one is null/empty)? Or am I missing something?

Reply:
You're right: either the client ID or the assigned client ID must be used, and the former only if the latter is null or empty.
In this case, however, this particular check is done by the call to ValidateConnectionAsync() above, which may (or may not) have modified connectPacket.ClientId [on line 308](https://github.com/chkr1011/MQTTnet/blob/1a41415dd165560cff97bb4ab52592133a013c55/Source/MQTTnet/Server/MqttClientSessionsManager.cs#L308).